src/meltano/cli/__init__.py

```diff
@@ -60,3 +58,23 @@ def main():
     except CliError as cli_error:
         cli_error.print()
         sys.exit(1)
+
+
+exit_code = None
```

Comment:
What do you think of moving to the top of the file, adding a comment to explain what this is and how it will be used, and then also adding a type hint.

Something like:

```py
# Hold the exit code value for error reporting during termination/cleanup:
exit_code: None | int = None
```